docs: Add docs for docker usage

karmab opened the pull request:
the corresponding docker image lives at docker.io/karmab/mgmt:0.0.14

purpleidea commented:
@karmab Thanks! Karim!

CI failed because we require a prefix for commits: eg: `docs: Add docs for docker usage` or something.
Could you instead add these docs to docs/docker-guide.md instead please?

Cheers!

purpleidea commented:
@karmab Sorry I did not see your update sooner. GH doesn't notify new pushes...

In any case, build failed because of an extra newline at the end of the dockerfile:

https://travis-ci.org/purpleidea/mgmt/jobs/334697863#L787

Do you mind fixing and updating please? Also can you move the Dockerfile into the docker/ folder?

Thank you! (Ping me here when it's done so I can merge)

purpleidea commented on Dockerfile.static:
```diff
@@ -0,0 +1,9 @@
+FROM centos:7
```
Can you move the docker files into docker/ folder please?

purpleidea commented on docs/quick-start-guide.md:
```diff
@@ -10,6 +10,9 @@ Once you're familiar with the general idea, please start hacking...
 ## Quick start
 
+### Using Docker
+check [docker-guide](docs/docker-guide.md) and skip to the *Running mgmt* section
```
Please use a full sentence with capitalization and period and move this section to the end after `Vagrant` server.

purpleidea commented on Makefile:
```diff
@@ -302,4 +302,16 @@ deb:
+build_container:
+	cd docker
+	docker build -t purpleidea/mgmgt-build -f ./Dockerfile.build .
+	docker run -td --name mgmt-build purpleidea/mgmgt-build
```
still has the typo...

karmab replied:
done

purpleidea commented on docs/quick-start-guide.md:
```diff
@@ -52,6 +52,10 @@ If you would like to avoid doing the above steps manually, we have prepared a
 project directory, run a `vagrant up`, and then a `vagrant status`. From there,
 you can `vagrant ssh` into the `mgmt` machine. The MOTD will explain the rest.
 
+### Using Docker
```
Should be `## Docker` instead. Note the existing headings.

karmab replied:
done

purpleidea commented on Makefile:
```diff
@@ -329,4 +329,16 @@ deb:
 
+build_container:
+	cd docker
+	docker build -t purpleidea/mgmt-build -f ./Dockerfile.build .
```
isn't the path `docker/Dockerfile.build` ? Same below

karmab commented:
@purpleidea 
- rebased on top of master
- fixed as per your requests
- tested successfully ( after adding additional dependencies in the building container) . the resulting latest binary can be found at docker.io/karmab/mgmt:06f2d65500b6e823b86fa45ec775a3b8f10c552c

purpleidea commented:
Merged, thanks!
